ros2action/ros2action/verb/send_goal.py

```diff
@@ -176,7 +188,18 @@ def _sigint_cancel_handler(sig, frame):
         print('Goal accepted with ID: {}\n'.format(bytes(goal_handle.goal_id.uuid).hex()))
 
         result_future = goal_handle.get_result_async()
-        rclpy.spin_until_future_complete(node, result_future, timeout_sec=timeout)
+        if timeout == 0:
+            # Not waiting for the result, just return immediately
+            rclpy.spin_until_future_complete(node, result_future, timeout_sec=0)
+        else:
+            elapsed_time = 0
+            while rclpy.ok() and not result_future.done():
+                # Spin until the result is available or timeout occurs
+                # To trigger the signal handler, it calls spin_until_future_complete with 1 second
+                elapsed_time += 1
+                rclpy.spin_until_future_complete(node, result_future, timeout_sec=1)
+                if timeout is not None and elapsed_time >= timeout:
+                    break
```

Comment:
the other thing we can do instead of this is that, we can move signal handler registration before `rclpy.init()`. so that it can call the deferred signal handlers in rclpy to trigger the guard condition to wake up the waitsets and shut down the default context gracefully. but this approach calls the `ros2action send_goal` signal handler **after** deferred signal handler shuts down the context. this does not allow us to check the cancel response result, because the context is already shut down. i believe that current PR is more appropriate process compared to that approach.